clients/venice-producer/src/main/java/com/linkedin/venice/producer/AbstractVeniceProducer.java

```diff
@@ -111,12 +110,11 @@ protected void configure(
   }
 
   private VeniceWriter<byte[], byte[], byte[]> getVeniceWriter(VersionCreationResponse versionCreationResponse) {
-    Properties writerProps = new Properties();
+    Properties writerProps = producerConfigs.getPropertiesCopy();
 
     if (versionCreationResponse.isEnableSSL()) {
       writerProps.put(KAFKA_OVER_SSL, "true");
       writerProps.put(SSL_KAFKA_BOOTSTRAP_SERVERS, versionCreationResponse.getKafkaBootstrapServers());
-      writerProps.put(KAFKA_SECURITY_PROTOCOL, producerConfigs.getString(KAFKA_SECURITY_PROTOCOL));
       writerProps.putAll(new SSLConfig(producerConfigs).getKafkaSSLConfig());
```

Comment:
I'm still a little confused to how the change is related to the PR title. If `writePros = producerConfigs.getPropertiesCopy();` then we can technically remove ` writerProps.putAll(new SSLConfig(producerConfigs).getKafkaSSLConfig());` too right?

Reply:
Yep. We can remove that too. Thanks for catching it.

The way it overrides it is via passing it through. Say we want to edit the default `delivery.ms`, we specify `kafka.delivery.ms` it in online `producerConfigs` object and it gets passed to the writer via `writerProps`. In that way, we can configure the pubsub system.

I can change the PR title and commit message to specify that we're only configuring the `VeniceWriter` instead of the pubsub producer.

Reply:
I see, now it makes sense.